### ISSUE TITLE
Gitops : Error creating : pods xxxx is forbidden 

### ISSUE THREAD

**Issue by Gilles-Plaquet** [IBM/cp4waiops-gitops#229]
While I was trying to deploy the Event and the AI-Manager I stumbled across an issue regarding permissions that results into a failed to create x.
I added a screenshot in the attachments regarding the error.
I already checked that my argo-cd has te required cluster-role bindings. Just to make sure, I added a screenshot of the yaml file  of this role binding aswell.

Hoping someone can help me resolve this issue !
Thanks in advance.

Kind regards,
Gilles

<img width="893" alt="Screenshot 2022-11-29 at 15 33 44" src="https://user-images.githubusercontent.com/96468037/204584370-56fe8f9b-5feb-4f5c-9221-327769dce7e7.png">
<img width="1096" alt="Screenshot 2022-11-29 at 17 03 25" src="https://user-images.githubusercontent.com/96468037/204584469-dcf6263f-bf1c-4378-988a-602f1fb81a93.png">

**Comment by gyliu513** [comment]
@Gilles-Plaquet can you provide more info, at which step you failed based on the document at here https://ibm.github.io/cp4waiops-gitops/docs/how-to-deploy-cp4waiops-35 ? What is your OCP version?

**Comment by Gilles-Plaquet** [comment]
@gyliu513 the current Openshift version is 4.8.39. In the documentation it stated that i should be above 4.5 so i guess that should be fine.

I was able to create the ceph-cluster, and the shared  application without any issues ( all app details seem to be healthy there)
I guess its the moment it started installing the AI-Manager. Then i started to get the issue stated above. 

Hope this information helped.

**Comment by gyliu513** [comment]
@Gilles-Plaquet can you login to your ocp cluster and run the command `oc get po -n cp4waiops` to check the pod status? If there are some pods not running, can you run `oc logs` for one of the not running pod and append the log here?

**Comment by Gilles-Plaquet** [comment]
@gyliu513  there are none.
in the error message he is telling that he can't create pods. thats why we don't see anything i think.

<img width="629" alt="Screenshot 2022-11-29 at 18 00 56" src="https://user-images.githubusercontent.com/96468037/204594194-4701e4a6-d90d-48f2-b275-74eafe958cd0.png">

I went to check in the openshift  interface and then i see this :
<img width="1128" alt="Screenshot 2022-11-29 at 18 03 45" src="https://user-images.githubusercontent.com/96468037/204594864-fbf60ac6-5550-4c4f-adb2-a350a7c4656f.png">

**Comment by Gilles-Plaquet** [comment]
@gyliu513 exactly, that was my reasoning as well... permission issue but i have all the cluster permissions. 
Thanks already for the help !

I also ran the command and everything in the rook-ceph cluster looks fine(too me) .
<img width="1481" alt="Screenshot 2022-11-29 at 18 34 28" src="https://user-images.githubusercontent.com/96468037/204601832-1f0f4f0e-7164-41ed-8424-33ee7ef98697.png">

**Comment by gyliu513** [comment]
@Gilles-Plaquet let me check more with @huang-cn and @morningspace , they are located in China, and hope we can give you more info tomorrow, thanks!

**Comment by Gilles-Plaquet** [comment]
thanks a lot already !

**Comment by morningspace** [comment]
@Gilles-Plaquet 
I see you mentioned that you are deploying both Event and the AI-Manager. May I know which install option you are taking, e.g.: to install it one by one, or use the all-in-one template. Also, may I know which release you deploy. Can you share the outputs of `oc get csv` under namespace `cp4waiops` and `ibm-common-services`?

**Comment by Gilles-Plaquet** [comment]
I used the one by one installation since, the other one was in technical preview. I opted for release 3.5. 

- oc get csv -n cp4waiops output:

![Screenshot 2022-11-30 at 08 58 08](https://user-images.githubusercontent.com/96468037/204739474-a94025ac-f3c5-4f9e-b0c9-02e5c72e2c44.png)

- oc get csv -n ibm-common-services

![Screenshot 2022-11-30 at 08 59 38](https://user-images.githubusercontent.com/96468037/204739759-bc6020e5-3f80-47c5-b0fd-02a487115931.png)

**Comment by Gilles-Plaquet** [comment]
@morningspace Thanks a lot! 
In case a webex,zoomcall,....  is easier to help solve the issue, that is possible ofcourse !

**Comment by huang-cn** [comment]
@Gilles-Plaquet I don't understand why  there's this `runAsUser: Invalid value: 1001` error appears here,  the AIOPS should not use `runAsUser` scc option at all, it shouldn't specify any UID value and let OCP to allocate one.  I'm wondering if the cagalog image in this env is the same as in ours?  
Could you run commands below to check the catalogsource image and operator scc settings?
```console
oc -n openshift-marketplace get catalogsource ibm-operator-catalog -oyaml|grep image:

oc -n cp4waiops get deploy iaf-core-operator-controller-manager -oyaml|grep -v 'f:securityContext'|grep securityContext  -A8

oc -n ibm-common-services get deploy ibm-common-service-operator -oyaml|grep -v 'f:securityContext'|grep securityContext  -A8
```

**Comment by Gilles-Plaquet** [comment]
- oc -n openshift-marketplace get catalogsource ibm-operator-catalog -oyaml|grep image:
<img width="895" alt="Screenshot 2022-12-01 at 14 26 16" src="https://user-images.githubusercontent.com/96468037/205064631-9983345a-703a-4a18-80b4-e26b2669ae04.png">

- oc -n cp4waiops get deploy iaf-core-operator-controller-manager -oyaml|grep -v 'f:securityContext'|grep securityContext  -A8
<img width="895" alt="Screenshot 2022-12-01 at 14 29 07" src="https://user-images.githubusercontent.com/96468037/205065073-e9f092dc-f0a5-4dc7-b771-c83763bff86e.png">

- oc -n ibm-common-services get deploy ibm-common-service-operator -o yaml|grep -v 'f:securityContext'|grep securityContext  -A8
<img width="885" alt="Screenshot 2022-12-01 at 14 32 00" src="https://user-images.githubusercontent.com/96468037/205065647-43e2dbbb-4f9c-40e3-b138-e5d39097fcc9.png">

Currently the namespace is not existing however yesterday it was, see the post above. this might be since openshift was unable to install the operator.

**Comment by gyliu513** [comment]
@Gilles-Plaquet the AIOps never uninstall `ibm-common-services` components unless you remove them manually, it is weird, we can talk next Monday to dig more, hope it is OK. Thanks!
